cterasdk/core/files/collaboration.py

```diff
@@ -0,0 +1,156 @@
+import logging
+from . import common
+from ..enum import ProtectionLevel, CollaboratorType, SearchType, PortalAccountType, FileAccessMode
+from ..types import UserAccount, GroupAccount
+from ...common import Object
+from ...exception import CTERAException
+
+
+def _search_collaboration_member(ctera_host, account, cloud_folder_uid):
+    search_param = Object()
+
+    if account.account_type == PortalAccountType.User:
+        search_param.searchType = SearchType.Users
+    elif account.account_type == PortalAccountType.Group:
+        search_param.searchType = SearchType.Groups
+    else:
+        raise CTERAException("Invalid account type", None, account_type=account.account_type)
+
+    search_param.searchTerm = account.name
+    search_param.resourceUid = cloud_folder_uid
+    search_param.countLimit = 1
+    response = ctera_host.execute('', 'searchCollaborationMembers', search_param)
+    if not response.objects:
+        logging.getLogger().warning('Could not find results that match your search criteria. %s', {'name': account.name})
+    elif len(response.objects) > 1:
+        logging.getLogger().warning('Search returned multiple results. Skipping...')
+    else:
+        collaborator = response.objects.pop()
+        if account.is_local and account.name == collaborator.name:
+            logging.getLogger().debug('Found local account. %s', {'name': account.name})
+            return collaborator
+        if not account.is_local and account.directory == collaborator.domain and account.name == collaborator.name:
+            logging.getLogger().debug('Found domain account. %s', {'name': account.name, 'domain': account.directory})
+            return collaborator
+    return None
+
+
+def share(ctera_host, path, recipients, as_project, allow_reshare, allow_sync):
```

Comment:
If I understand correctly, this method overrides the current configuration. So, if the file is already shared, the previous configurations will be deleted, right? If so, is this the intended behavior?